### PR TITLE
Feature/household api integration

### PR DESCRIPTION
**Pull request by DaQuirm**
This should take care of polling while discovering/deleting devices and various related issues :cat: 

@larsblumberg @ghophp

### CODE REVIEW

File: public/app/common/helpers/device/device-helpers.js [hunk 1]
```diff
@@ -33,6 +34,13 @@ var Helpers = {
     return component.properties.capabilities[0];
   },
 
+  getDeviceUrl: function getDeviceUrl (device) {
+    var link = _.find(device.links, function getLinkToSelf (linkObject) {
+      return linkObject.rel[0] === 'self';
+    });
+    return link.href;
+  },
+
```
Comment (larsblumberg) [inline public/app/common/helpers/device/device-helpers.js]: Would avoiding a variable and thus writing `_.find(...).href` be less readable here?

Comment (DaQuirm) [inline public/app/common/helpers/device/device-helpers.js]: A bit messier perhaps :cat:

File: public/app/common/helpers/poll/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+function Poll (options) {
+
+  this.options = options;
+  this.options.predicate = this.options.predicate || function truthSayer () { return true; };
```
Comment (larsblumberg) [inline public/app/common/helpers/poll/index.js]: As `truthSayer` is a noun (and thus a class name), we could call the method here something like `forever` or `alwaysTrue`

Comment (DaQuirm) [inline public/app/common/helpers/poll/index.js]: Sorry, I went a little bit functional, where it's okay to use nouns for function names: `mapper`, `reducer` and so forth. I'd argue that you can extend this notion to daily JavaScript :smile_cat: 
You shouldn't use nouns for method names, but I'd say that functions are often more "nouns" than "verbs".
Consider this Promise API [article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) from MDN: the function is called `executor` to give just one example.
```js
new Promise(executor)
```

File: public/app/common/services/devices/discovery-service/discovery-service.js [hunk 3]
```diff
@@ -1,80 +1,72 @@
...
-        return Rx.Observable.fromPromise(
-          fetch(discoveryUrl));
-      }).flatMap((discoveryStatus) => {
-        return Rx.Observable.create(handleDiscoveryStatus(discoveryStatus, subject));
-      });
+  stopDiscovery: function stopDiscovery () {
+    sessionPoll.cancel();
   }
+
 };
```
Comment (larsblumberg) [inline public/app/common/services/devices/discovery-service/discovery-service.js]: The discovery session code reads very well! 😻

File: public/app/common/services/devices/discovery-service/test/discovery-service.spec.js [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
+    promiseHelper = require('helpers/promise');
 
+require('whatwg-fetch');
 
 var SessionState = require('../discovery-session-state.js');
+var UrlHelpers = require('helpers/url');
 var mock = require('./discovery-service.mock.js');
 
-var apiHost = 'https://householdmockapi-dev.yetu.me';
+var apiHost = '/household';
```
Comment (larsblumberg) [inline public/app/common/services/devices/discovery-service/test/discovery-service.spec.js]: `apiHost` isn't a host anymore, so rename it to `householdUrl` maybe?

File: public/app/stores/discovery-store/test/discovery-store.spec.js [hunk 5]
```diff
@@ -1,33 +1,18 @@
...
-      expect(discoveryStore.discoveryData.model).toEqual({});
-      expect(discoveryStore.discoveryData.error).toEqual({});
-    });
-  });
+describe('Discovery Store', function () {
 
   describe('getInitialState', function () {
-    it('returns initial model', function () {
-      expect(discoveryStore.getInitialState()).toEqual(discoveryStore.discoveryData);
+    it('returns an empty object', function () {
```
Comment (larsblumberg) [inline public/app/stores/discovery-store/test/discovery-store.spec.js]: Given that initial state is empty, is this test still useful?

File: public/app/common/services/devices/discovery-service/discovery-service.js (hunk 3 above)
Comment (DaQuirm) [inline public/app/common/services/devices/discovery-service/discovery-service.js]: Thank you! :smile_cat:

File: public/app/common/helpers/poll/index.js (hunk 2 above)
Comment (larsblumberg) [inline public/app/common/helpers/poll/index.js]: I see, thanks for clarifying that!

File: public/app/common/services/devices/discovery-service/test/discovery-service.spec.js (hunk 4 above)
Comment (DaQuirm) [inline public/app/common/services/devices/discovery-service/test/discovery-service.spec.js]: Well spotted, I'll go with `householdApiUrl`.

File: public/app/stores/discovery-store/test/discovery-store.spec.js (hunk 5 above)
Comment (DaQuirm) [inline public/app/stores/discovery-store/test/discovery-store.spec.js]: Nah, it's not very useful, IMHO.

**Comment by larsblumberg** [comment]
LGTM

**Pull request merged**